lib/authorizer/oauth2.js

```diff
@@ -58,6 +59,34 @@ module.exports = {
      * @param {AuthHandlerInterface~authPreHookCallback} done -
      */
     pre: function (auth, done) {
+        const authId = auth.get('authId'),
+            refreshTokenHelper = auth.get('refreshTokenHelper');
+
+        // Go into the refresh token flow ONLY IF we have all necessary information present
+        if (authId && refreshTokenHelper && _.isFunction(refreshTokenHelper.refreshToken)) {
+            const refreshTokenPromise = refreshTokenHelper.refreshToken(authId);
+
+            return Promise.race([
+                refreshTokenPromise,
+                new Promise((resolve) => {
+                    // We add a timeout to the refresh token promise to ensure that the request is not
+                    // hung up on the refresh token call. If the refresh token call takes more than
+                    // REFRESH_TOKEN_TIMEOUT_MS, we abandon and move forward with the execution.
+                    setTimeout(() => {
+                        return resolve();
+                    }, REFRESH_TOKEN_TIMEOUT_MS);
+                })
+            ]).then((accessToken) => {
+                accessToken && auth.set('accessToken', accessToken, true);
+            }).catch(() => {
+                // do nothing, we throw an error from app when refresh fails
+                // ? TODO: should we throw an error here? app handles logging to console
+                // ? anyway
```

Comment:
Think about how should someone using `postman-runtime` as a library would handle it. How would `newman` handle it?